calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java

```diff
@@ -100,83 +103,66 @@ public void importStreamFromDelimitedFile(
-				if (durationminutes < 10)
-				{
+                                if ( component.getProperty("SUMMARY") == null )
+                                {
+                                    M_log.warn("IcalendarReader: SUMMARY is required; event not imported");
+                                    continue;
+                                }
+                                DateTime from = new DateTime(Date.from(ZonedDateTime.now().minusMonths(6).toInstant()));
+                                DateTime to = new DateTime(Date.from(ZonedDateTime.now().plusMonths(12).toInstant()));
+                                Period range = new Period(from, to);
+                                
+				PeriodList list = component.calculateRecurrenceSet(range);
+				for (Iterator j = list.iterator(); j.hasNext();) 
+                                {
+                                    Period period = (Period) j.next();
+                                    Dur duration = period.getDuration();
+                                    int durationminutes = duration.getMinutes();
+                                    int durationhours = duration.getHours();
+                                    //todo investiage ical4j's handling of 'days'
+                                    
+                                    if (durationminutes < 10)
+                                    {
 					durationformat = "0"+durationminutes;
-				}
-				else
-				{
+                                    }
+                                    else
+                                    {
 					durationformat = ""+durationminutes;
-				}
+                                    }
 
-				if (durationhours != 0)
-				{
+                                    if (durationhours != 0)
+                                    {
 					durationformat = durationhours+":"+durationformat;
-				}
-				
-				String description = "";
-				if ( component.getProperty("DESCRIPTION") != null )
-					description = component.getProperty("DESCRIPTION").getValue();
-               
-            String location = "";
-				if (component.getProperty("LOCATION") != null)
-               location = component.getProperty("LOCATION").getValue();
-               
-				String columns[]	= 
-						{component.getProperty("SUMMARY").getValue(),
-						 description,
-						 DateFormat.getDateInstance(DateFormat.SHORT, rb.getLocale()).format(dtstartdate.getDate()),
-						 DateFormat.getTimeInstance(DateFormat.SHORT, rb.getLocale()).format(dtstartdate.getDate()),
-						 durationformat,
-						 location};
-				
-				// Remove trailing/leading quotes from all columns.
-				//trimLeadingTrailingQuotes(columns);
-			
-				handler.handleRow(
-					processLine(
-						columnDescriptionArray,
-						lineNumber,
-						columns));
-					
-				lineNumber++;
-			
+                                    }
+                                    String description = "";
+                                    if ( component.getProperty("DESCRIPTION") != null ) {
```

Comment:
I don't _think_ this is needed.  getProperty() returns a "Property" type (or null), not a String.  If it's null then it's ok for the "description" string to be empty. Also, FWIW this is not new code, only the indenting has changed.